wgengine/router/router_linux.go

```diff
@@ -1548,6 +1580,36 @@ func checkIPRuleSupportsV6(logf logger.Logf) error {
 	return netlink.RuleAdd(rule)
 }
 
+// Checks if the running openWRT system is using mwan3, based on the heuristic
+// of the config file being present as well as a policy rule with a specific
+// priority (2000 + 1 - first interface mwan3 manages) and mark (1<<8 - also
+// symbolizing the first interface).
+func checkOpenWRTUsingMWAN3() (bool, error) {
+	if _, err := os.Stat("/etc/config/mwan3"); err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	rules, err := netlink.RuleList(netlink.FAMILY_V4)
+	if err != nil {
+		return false, err
+	}
+	for _, r := range rules {
+		// We want to match on a rule like this:
+		//    2001:	from all fwmark 0x100/0x3f00 lookup 1
+		//
+		// We dont match on the mask because it can vary, or the
+		// table because I'm not sure if it can vary.
```

Comment:
If the mask can vary, that suggests the match value might also change, e.g. if I set the mask to 0xff, would the match value be 0x10 ?

If the mark numbers can definitely be configured, I'd stick to checking the table numbers, and maybe a non-zero r.Mark?

Reply:
Yeah, it seems they do some horrible calculation (in bash) to adjust the interface ID to fit into the mask bits you selected. I'll just check the mark is non-zero.